### ISSUE TITLE
postgres create table with storage parameters parse error

### ISSUE THREAD

**Issue by bpfaust** [sqlfluff/sqlfluff#4297]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Always getting a parse error on create tables (temp or perm) when using "with (storage_parameters = value)" syntax (https://www.postgresql.org/docs/13/sql-createtableas.html)

### Expected Behaviour

Below should pass:

```sql
create temp table a_new_table as
with (appendoptimized = true, compresstype = zstd) as
select
    column_1
    , column_2
    , column_3
from schema.tablename
group by 1, 2, 3;
```

### Observed Behaviour

Below returned with "FAIL" status

`==== parsing violations ====
L:   3 | P:   1 |  PRS | Line 3, Position 1: Found unparsable section: 'create temp table a_new_table\nwith (a...'`

Also returns the same when \n removed:

`L:   3 | P:   1 |  PRS | Line 3, Position 1: Found unparsable section: 'create temp table a_new_table with (appe...'`

### How to reproduce

Run `lint` or `parse` on the above query

### Dialect

postgres

### Version

sqlfluff 1.4.5
python 3.11.1

### Configuration

[sqlfluff_config.txt](https://github.com/sqlfluff/sqlfluff/files/10447363/sqlfluff_config.txt)

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by bpfaust** [comment]
Realized there's a spurious "as" keyword in the expected behavior sql, line 1.  That is an artifact from testing what the unparsable section was when I was commenting out the "with" line and should be ignored.  Following SQL still generates the parse error:

```sql
create temp table a_new_table
with (appendoptimized = true, compresstype = zstd) as
select
    column_1
    , column_2
    , column_3
from schema.tablename
group by 1, 2, 3;
```

**Comment by AidanHarveyNelson** [comment]
Hi I am writing a PR to allow multiple delimited storage parameters.

Not sure if you are aware of this and the example you have given is just a sample but I am getting invalid Syntax when running the above SQL command. As per the documentation https://www.postgresql.org/docs/13/sql-createtable.html#SQL-CREATETABLE-STORAGE-PARAMETERS none of the parameters above are listed as valid. Also only integers, floating points and boolean values are allowed in the value section. So `zstd` would not be considered valid.

Can you please help me understand and show me the above command not throwing a syntax error?

![image](https://user-images.githubusercontent.com/79121034/216725049-b92d815c-a086-497b-be55-a0d49b1d0129.png)

![image](https://user-images.githubusercontent.com/79121034/216725069-763597fd-cf14-4c50-a88f-6be22c298dc8.png)

**Comment by bpfaust** [comment]
Sure - was afraid this might be the case.  Our company uses a Greenplum database so it's always murky which parts of our queries are "stock" postgres vs. GP-specific...those parameters are basically setting the table up as column-oriented and then which compression algorithm to use on it.

I can dig around in their docs to look for allowed keywords...or do you think this may warrant a new dialect?

**Comment by bpfaust** [comment]
Found them: https://docs.vmware.com/en/VMware-Tanzu-Greenplum/6/greenplum-database/GUID-ref_guide-sql_commands-CREATE_TABLE_AS.html

**Comment by AidanHarveyNelson** [comment]
@greg-finley Is the Postgres dialect only for `stock` postgres or is meant to support some more specific ones such as Greenplum which is built off Postgres?

If this would fall under the requirements of needing a new dialect I am happy to make a start on it

**Comment by greg-finley** [comment]
@AidanHarveyNelson I haven't fully digested the issue at hand here, but general advice is, if it's addable to Postgres _without breaking other Postgres stuff_, feel free to just add it there (even if not truly a thing supported in Postgres). But if it's truly incompatible, starting a new dialect extending from Postgres makes sense.

Reading this for just a few minutes, it seems it could just be added to Postgres?

**Comment by AidanHarveyNelson** [comment]
It can definitely be added without breaking other Postgres stuff. Thanks for confirming the spirit of the dialect.
